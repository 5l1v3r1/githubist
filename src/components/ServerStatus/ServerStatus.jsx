// @flow

import React, { type Node } from 'react';
import { Route } from 'react-router-dom';

type Props = {
  code: number,
  children: Node,
};

const Status = ({ code, children }: Props) => (
  <Route
    render={({ staticContext }) => {
      if (staticContext) {
        /* eslint-disable no-param-reassign */
        /* $FlowIgnoreNextLine */
        staticContext.serverStatusCode = code;
        /* eslint-enable no-param-reassign */
      }

      return children;
    }}
  />
);

export default Status;
