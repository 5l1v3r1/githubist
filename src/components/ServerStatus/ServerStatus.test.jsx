// @flow

import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import ServerStatus from './ServerStatus';

it('renders correctly', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <ServerStatus code={404}>Not Found</ServerStatus>
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});
