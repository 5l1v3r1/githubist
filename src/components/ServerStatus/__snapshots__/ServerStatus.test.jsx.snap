// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Status
      code={404}
    >
      <Route
        render={[Function]}
      >
        Not Found
      </Route>
    </Status>
  </Router>
</MemoryRouter>
`;
